### PR TITLE
Update `/api/comments/byepisodeid` to handle episode without a `socialInteract`

### LINKED ISSUE #265: Episode with invalid socialInteract not handled properly
In the `/api/comments/byepisodeid` handler, if the episode doesn't have a `socialInteract`, the invalid response is not sent.

### PR DESCRIPTION
**Pull request by stevencrader**
Resolves #265

### CODE REVIEW

**Comment by stevencrader** [comment]
@dellagustin can you look at this?

**Comment by dellagustin** [comment]
Hello @stevencrader , how did you test this? Directly calling the `/api/comments/byepisodeid` endpoint?
As far as I know, if `socialInteract` is not used by an episode, the _Show comments_ button will not be shown.
Of course, it should be fixed regardless, but I just wanted to know if this is actually a bug observed during a regular user journey, or something that is only a "sanity check" fix.

**Comment by stevencrader** [comment]
Hi @dellagustin. I did call it directly.

I noticed there was a typo in length and wanted to verify it was a real issue and then test my fix.

I agree that the logic before won't call this if there isn't a socialInteract tag do it isn't that big of an issue.

**Comment by dellagustin** [comment]
It should fix https://github.com/Podcastindex-org/web-ui/issues/232 as well, same as https://github.com/Podcastindex-org/web-ui/issues/265.
I'll run a quick test.

File: server/index.js [hunk 1]
```diff
@@ -133,10 +133,11 @@ app.use('/api/comments/byepisodeid', async (req, res) => {
 
     const socialInteract = response.episode.socialInteract && response.episode.socialInteract.filter((si) => si.protocol === 'activitypub');
 
-    if(!socialInteract && socialInteract.lenght >= 0) {
+    if(socialInteract === undefined || socialInteract.length >= 0) {
```
Comment (dellagustin) [inline server/index.js:136]:
```suggestion
    if(socialInteract === undefined || socialInteract.length === 0) {
```

The original condition was broken in more then just the typo, I think, but if I not mistaken, it was designed for: 
1. Returning 400 when the episode does not have a socialInteract (at all)
2. Returning 400 when the episode has one or more socialInteract, but none is `activitpub` (possible, even though unlikely at the moment)

With the code you proposed, the valid use case (episodes with activity pub comments) would not work, as socialInteract.length >= 0 would be true, thus also triggering the 400 error.

I tested the suggestion in this comment and it worked with both episodes with and without comments.
I could not test the scenario described with 2, as I am not aware of any podcast with that use case. I would need to fabricate one, and that would be time consuming.

**Review by dellagustin — CHANGES_REQUESTED** [review]

**Pull request merged**